### DISCUSSION TITLE
Connection to modbus failed

### DISCUSSION THREAD

**Discussion by xanak76** [nathanmarlor/foxess_modbus#227] [Support] [answered]
Dear all,
I have configured either the Elfin EW11 or the USR TCP232 410s according to the instructions in the wiki and connected them to the RS485 interface of my H3 inverter (port 1 and 2). Unfortunately, I get the following error message with both adapters when setting up the FoxESS modbus integration in Home Assistant:

"Unable to connect to your adapter. Ensure that the adapter is properly configured (see the setup link above) and the connection details are correct, then try again. Details: Modbus Error: [Connection] ModbusTcpClient(192.168.0.81:502): Connection unexpectedly closed 5.745887756347656e-05 seconds into read of 8 bytes without response from unit before it closed connection."

I would be very grateful if someone could help me with this problem.

Thanks and greetings

fred76

**Comment by calum-mcfarlane** [comment]
Have you tried doing any testing with modpoll? Also worth double checking the continuity of the connections from the terminals of the inverter plug to the terminals of the adapter unit (whether that be the Efin or the USR).

**Comment by canton7** [comment]
It does sound like HA is connecting to your adapter OK, but the adapter isn't getting a response back from the inverter.

I'd start by checking your wiring: correct polarity? Connected to the right pins? Proper electrical connection to the pins? Some people like to use Cat5 which has a habit of snapping when screwed.

**Comment by alanmcgore** [comment]
The Elfin has a status page when you login which should say how much has been transmitted both over serial and TCP along with any failures.

**Comment by xanak76** [comment]
Dear all,
thank you very much for your replies. I will double check the wiring. I used cat7 (laying cable, unfortunately i don't know the correct english expression). It is very stiff and might be broken at the plug of the inverter. However, i will try cat7 patch cable, which is softer. Is it correct to use a pair (of twisted pair, e.g. red and white/blue and white) of cables for port 1 and 2 of the plug? 
I will get back when i changed the wiring,

Thanks a lot and best regards

fred

**Reply by canton7** [reply]
For a H3 yes it's pins 1 and 2.

If it's a short length of cable, having a twisted pair doesn't really matter - I'm using speaker wire for a run of ~1m which is fine. For a longer run yes you should use two parts of a twisted pair.

If you can, use a multimeter for check for electrical continuity between the pin of the connector and the other end of the cable.

**Comment by canton7** [comment]
@xanak76 Did you ever get to the bottom of this?

**Reply by xanak76** [reply, marked as answer]
Dear all, please excuse my late feedback. I only got around to checking the wiring today. In fact, this was the problem. By using a twisted pair cable pair per pin, everything has now worked out best. Thank you for your support and the great plugin.

Kind regards xanak76

**Reply by canton7** [reply]
Thanks for the update, glad you got it working!

**Reply by xanak76** [reply]
Thank You! :)
